fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/FromItem.java

```diff
@@ -1,22 +1,84 @@
 /*
- * (C) Copyright IBM Corp. 2019
+ * (C) Copyright IBM Corp. 2019,2021
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 package com.ibm.fhir.database.utils.query;
 
+import com.ibm.fhir.database.utils.query.expression.StatementRenderer;
+
 /**
- * An entry in the FROM clause
+ * An element in the FROM clause
  */
 public class FromItem {
+
+    // The source we are joining to
+    private final RowSource rowSource;
+
+    // Optional alias...but recommended to avoid nasty surprises
     private final Alias alias;
 
-    protected FromItem(Alias alias) {
+    /**
+     * @param rowSource
+     * @param alias
+     */
+    protected FromItem(RowSource rowSource, Alias alias) {
+        this.rowSource = rowSource;
         this.alias = alias;
     }
 
+    protected FromItem(RowSource rowSource) {
+        this.rowSource = rowSource;
+        this.alias = null;
+    }
+
+    /**
+     * Get the alias associated with this item in the from clause.
+     * @return the alias for this item. Can be null
+     */
     public Alias getAlias() {
+        // can be null
         return this.alias;
     }
+
+    /**
+     * Is this an ANSI-style join?
+     * @return
+     */
+    public boolean isAnsiJoin() {
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        return toPrettyString(false);
+    }
+
+    public String toPrettyString(boolean pretty) {
```

Comment:
feels like this should be extracted to an interface?

Reply:
need more info

Reply:
it was used in other spots...  it feels fairly common in other classes.

Reply:
FromItem is a base class for FromJoin which overrides toPrettyString. I'm not seeing a use-case for an interface at this point. We can refactor if such a case comes up in the future.